### PR TITLE
feat: OptimizelyConfigService

### PR DESCRIPTION
**Pull request by oakbani**
## Summary
Adds support for OptimizelyConfigService. 

## Test plan
- Tested manually with FSC tests for getOptimizelyConfig
- Added unit tests for OptimizelyConfigService and new entities. 

## Issues
- "THING-1234" or "Fixes #123"

### CODE REVIEW

**Comment by zashraf1985** [comment]
@aliabbasrizvi It would be great if you can spare some time to review this.

**Comment by aliabbasrizvi** [comment]
Reviewing this now.

File: src/Optimizely/OptimizelyConfig/OptimizelyConfigService.php [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                    $variable->getType(),
+                    $variable->getDefaultValue()
+                );
+
+                $variablesKeyMap[$variableKey] = $optVariable;
+                $variablesIdMap[$variableId] = $optVariable;
+            }
+
+            $featureKey = $feature->getKey();
+            $this->featKeyOptlyVariableKeyVariableMap[ $featureKey] = $variablesKeyMap;
```
Comment (jaeopt) [inline src/Optimizely/OptimizelyConfig/OptimizelyConfigService.php]: Remove space in front of $featureKey

File: src/Optimizely/OptimizelyConfig/OptimizelyConfig.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * limitations under the License.
+ */
+namespace Optimizely\OptimizelyConfig;
+
+class OptimizelyConfig implements \JsonSerializable
+{
+    /**
+     * @var string Revision of the config.
+     */
+    private $revision;
```
Comment (aliabbasrizvi) [inline src/Optimizely/OptimizelyConfig/OptimizelyConfig.php]: I feel there should be a leading underscore in all private variables names

File: src/Optimizely/OptimizelyConfig/OptimizelyConfigService.php [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+use Optimizely\Config\ProjectConfigInterface;
+use Optimizely\Entity\Experiment;
+use Optimizely\Entity\Variation;
+
+class OptimizelyConfigService
+{
+    /**
+     * @var array List of experiments in config.
+     */
+    private $experiments;
```
Comment (aliabbasrizvi) [inline src/Optimizely/OptimizelyConfig/OptimizelyConfigService.php]: Same feedback as above

**Comment by oakbani** [comment]
@aliabbasrizvi  Thanks for reviewing. The latest PSR standard for PHP discourages to put a leading underscore before private/protected properties. 

> Property names MUST NOT be prefixed with a single underscore to indicate protected or private visibility. That is, an underscore prefix explicitly has no meaning.

https://www.php-fig.org/psr/psr-12/

**Comment by aliabbasrizvi** [comment]
Thanks for pointing that out @oakbani

**Comment by aliabbasrizvi** [comment]
Going to merge this.

**Pull request merged**